projects/gateway2/proxy_syncer/proxy_syncer.go

```diff
@@ -588,23 +589,32 @@ func (s *ProxySyncer) Start(ctx context.Context) error {
 				logger.Debug("context done, stopping proxy syncer")
 				return
 			case <-timer.C:
+				// build a list of reports to feed the status plugins
+				var proxiesWithReports []translatorutils.ProxyWithReports
+				var snapPlugins []registry.PluginRegistry
+
 				snaps := s.mostXdsSnapshots.List()
+
 				for _, snapWrap := range snaps {
-					var proxiesWithReports []translatorutils.ProxyWithReports
 					proxiesWithReports = append(proxiesWithReports, snapWrap.proxyWithReport)
+					snapPlugins = append(snapPlugins, snapWrap.pluginRegistry)
 
+					// init the snapshot status plugins
```

Comment:
nit: Not an actual issue, but we may want to pass pointers here so we don't get subtle issues if non-pointer/interface fields are added to PluginRegistry.